plugins/modules/fabric_transits_workflow_manager.py

```diff
@@ -0,0 +1,1354 @@
+        Description:
+            Check the fabric_transits. If the name is not available, return with a msg by setting the
+            self.status as 'failed' and return self.
+            Call the 'get_want_fabric_transits' function to collect the information
+            about the fabric transit passed by the user.
+        """
+
+        fabric_transits = config.get("sda_fabric_transits")
+        if not fabric_transits:
+            self.msg = "The parameter 'sda_fabric_transits' is missing under the 'config'."
+            self.status = "failed"
+            return self
+
+        self.get_want_fabric_transits(fabric_transits).check_return_status()
+
+        self.log("Desired State (want): {requested_state}".format(requested_state=self.want), "INFO")
+        self.msg = "Successfully retrieved details from the playbook"
+        self.status = "success"
+        return self
+
+    def get_task_details_by_id(self, task_id):
+        """
+        Get the details of a specific task in Cisco Catalyst Center.
+
+        Args:
+            self (object): An instance of a class that provides access to Cisco Catalyst Center.
+            task_id (str): The unique identifier of the task for which you want to retrieve details.
+        Returns:
+            dict or None: A dictionary containing detailed information about the specified task,
+            or None if the task with the given task_id is not found.
+        Description:
+            Call the API 'get_task_details_by_id' to get the details along with the
+            failure reason. Return the details.
+        """
+
+        result = None
+        response = self.dnac._exec(
+            family="task",
+            function="get_task_details_by_id",
+            params={"id": task_id}
+        )
+        self.log('Task Details: {response}'.format(response=response), 'DEBUG')
+        if response and isinstance(response, dict):
+            result = response.get('response')
+
+        return result
+
+    def get_tasks_by_id(self, task_id):
+        """
+        Get the tasks of a task ID in Cisco Catalyst Center.
+
+        Args:
+            self (object): An instance of a class that provides access to Cisco Catalyst Center.
+            task_id (str): The unique identifier of the task for which you want to retrieve details.
+        Returns:
+            dict or None: A dictionary status information about the specified task,
+            or None if the task with the given task_id is not found.
+        Description:
+            Call the API 'get_tasks_by_id' to get the status of the task.
+            Return the details along with the status of the task.
+        """
+
+        result = None
+        response = self.dnac._exec(
+            family="task",
+            function="get_tasks_by_id",
+            params={"id": task_id}
+        )
+        self.log('Task Details: {response}'.format(response=response), 'DEBUG')
```

Comment:
if not isinstance(response, dict):
        self.log("Failed to retrieve task details for task ID: {}".format(task_id), "ERROR")
        return task_status